File: server/src/model/entry.rs
use serde::{Deserialize, Serialize};
use diesel::{self, Queryable};

#[derive(Clone,Deserialize,Serialize,Queryable)]
pub struct Entry {
    pub id: i32,
    pub name: String,
    pub price: f64,
    pub voltage: f64,
    pub volume: f64,
    pub verified: bool,
    pub photo: String
}

File: src/types/entry.ts
export type Entry = {
  id: string
  user_id: string
  name: string
  price: number
  voltage: number
  volume: number
  verified: boolean | null
  photo: string | null
  label: string[]
}

export type NewEntry = Pick<Entry, 'voltage' | 'name' | 'price' | 'volume' | 'photo' | 'label'>
